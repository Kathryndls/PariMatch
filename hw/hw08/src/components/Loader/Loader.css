@-webkit-keyframes ldio-lgz85uwngd {
  0% {
    -webkit-transform: rotate(0);
            transform: rotate(0);
  }
  100% {
    -webkit-transform: rotate(360deg);
            transform: rotate(360deg);
  }
}

@keyframes ldio-lgz85uwngd {
  0% {
    -webkit-transform: rotate(0);
            transform: rotate(0);
  }
  100% {
    -webkit-transform: rotate(360deg);
            transform: rotate(360deg);
  }
}

.ldio-lgz85uwngd div {
  -webkit-box-sizing: border-box !important;
          box-sizing: border-box !important;
}

.ldio-lgz85uwngd > div {
  position: absolute;
  width: 72px;
  height: 72px;
  top: 14px;
  left: 14px;
  border-radius: 50%;
  border: 8px solid #000;
  border-color: #2b7de9 transparent #2b7de9 transparent;
  -webkit-animation: ldio-lgz85uwngd 1s linear infinite;
          animation: ldio-lgz85uwngd 1s linear infinite;
}

.ldio-lgz85uwngd > div:nth-child(2) {
  border-color: transparent;
}

.ldio-lgz85uwngd > div:nth-child(2) div {
  position: absolute;
  width: 100%;
  height: 100%;
  -webkit-transform: rotate(45deg);
          transform: rotate(45deg);
}

.ldio-lgz85uwngd > div:nth-child(2) div:before, .ldio-lgz85uwngd > div:nth-child(2) div:after {
  content: "";
  display: block;
  position: absolute;
  width: 8px;
  height: 8px;
  top: -8px;
  left: 24px;
  background: #2b7de9;
  border-radius: 50%;
  -webkit-box-shadow: 0 64px 0 0 #2b7de9;
          box-shadow: 0 64px 0 0 #2b7de9;
}

.ldio-lgz85uwngd > div:nth-child(2) div:after {
  left: -8px;
  top: 24px;
  -webkit-box-shadow: 64px 0 0 0 #2b7de9;
          box-shadow: 64px 0 0 0 #2b7de9;
}

.loadingio-spinner-dual-ring-cqkmvary9bi {
  width: 50px;
  height: 50px;
  display: inline-block;
  overflow: hidden;
  background: transparent;
}

.ldio-lgz85uwngd {
  width: 100%;
  height: 100%;
  position: relative;
  -webkit-transform: translateZ(0) scale(0.5);
          transform: translateZ(0) scale(0.5);
  -webkit-backface-visibility: hidden;
          backface-visibility: hidden;
  -webkit-transform-origin: 0 0;
          transform-origin: 0 0;
  /* see note above */
}

.ldio-lgz85uwngd div {
  -webkit-box-sizing: content-box;
          box-sizing: content-box;
}
/*# sourceMappingURL=Loader.css.map */