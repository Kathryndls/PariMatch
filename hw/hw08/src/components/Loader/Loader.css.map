{
    "version": 3,
    "mappings": "AAAA,UAAU,CAAV,eAAU;EACP,EAAE;IAAG,SAAS,EAAE,SAAS;;EACzB,IAAI;IAAG,SAAS,EAAE,cAAc;;;;AAEnC,AAAA,gBAAgB,CAAC,GAAG,CAAC;EAAE,UAAU,EAAE,UAAU,CAAA,UAAU;CAAG;;AAC1D,AAAA,gBAAgB,GAAG,GAAG,CAAC;EACpB,QAAQ,EAAE,QAAQ;EAClB,KAAK,EAAE,IAAI;EACX,MAAM,EAAE,IAAI;EACZ,GAAG,EAAE,IAAI;EACT,IAAI,EAAE,IAAI;EACV,aAAa,EAAE,GAAG;EAClB,MAAM,EAAE,cAAc;EACtB,YAAY,EAAE,uCAAuC;EACrD,SAAS,EAAE,kCAAkC;CAC/C;;AACD,AAAA,gBAAgB,GAAG,GAAG,AAAA,UAAW,CAAA,CAAC,EAAE;EAAE,YAAY,EAAE,WAAY;CAAE;;AAClE,AAAA,gBAAgB,GAAG,GAAG,AAAA,UAAW,CAAA,CAAC,EAAE,GAAG,CAAC;EACrC,QAAQ,EAAE,QAAQ;EAClB,KAAK,EAAE,IAAI;EACX,MAAM,EAAE,IAAI;EACZ,SAAS,EAAE,aAAa;CAC1B;;AACD,AAAA,gBAAgB,GAAG,GAAG,AAAA,UAAW,CAAA,CAAC,EAAE,GAAG,AAAA,OAAO,EAAE,gBAAgB,GAAG,GAAG,AAAA,UAAW,CAAA,CAAC,EAAE,GAAG,AAAA,MAAM,CAAC;EAC3F,OAAO,EAAE,EAAE;EACX,OAAO,EAAE,KAAK;EACd,QAAQ,EAAE,QAAQ;EAClB,KAAK,EAAE,GAAG;EACV,MAAM,EAAE,GAAG;EACX,GAAG,EAAE,IAAI;EACT,IAAI,EAAE,IAAI;EACV,UAAU,EAAE,OAAO;EACnB,aAAa,EAAE,GAAG;EAClB,UAAU,EAAE,kBAAkB;CAChC;;AACD,AAAA,gBAAgB,GAAG,GAAG,AAAA,UAAW,CAAA,CAAC,EAAE,GAAG,AAAA,MAAM,CAAC;EAC3C,IAAI,EAAE,IAAI;EACV,GAAG,EAAE,IAAI;EACT,UAAU,EAAE,kBAAkB;CAChC;;AACD,AAAA,wCAAwC,CAAC;EACtC,KAAK,EAAE,IAAI;EACX,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,YAAY;EACrB,QAAQ,EAAE,MAAM;EAChB,UAAU,EAAE,WAAW;CACzB;;AACD,AAAA,gBAAgB,CAAC;EACd,KAAK,EAAE,IAAI;EACX,MAAM,EAAE,IAAI;EACZ,QAAQ,EAAE,QAAQ;EAClB,SAAS,EAAE,aAAa,CAAC,UAAU;EACnC,mBAAmB,EAAE,MAAM;EAC3B,gBAAgB,EAAE,GAAG;EAAE,oBAAoB;CAC7C;;AACD,AAAA,gBAAgB,CAAC,GAAG,CAAC;EAAE,UAAU,EAAE,WAAW;CAAI",
    "sources": [
        "Loader.scss"
    ],
    "names": [],
    "file": "Loader.css"
}