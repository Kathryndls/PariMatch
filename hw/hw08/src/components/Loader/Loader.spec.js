import React from "react";
import Loader from "./Loader";

const setUp = (props) => shallow(<Loader {...props} />);
const setUpRender = (props) => render(<Loader {...props} />);

describe('should render Loader component', () => {
   let component;
   beforeEach(() => {
      component = setUp();
   })

   it ('Loader component should have a loadingio-spinner-dual-ring-cqkmvary9bi class', () => {
      const wrapper = component.find('.loadingio-spinner-dual-ring-cqkmvary9bi');
      expect(wrapper.length).toBe(1);
   });

   it ('Loader component should have a ldio-lgz85uwngd class', () => {
      const wrapper = component.find('.ldio-lgz85uwngd');
      expect(wrapper.length).toBe(1);
   });

   it ('Loader comonent should have a div tag', () => {
      const wrapper = component.find('div');
      expect(wrapper.length).toBe(5);
   });
});

describe('Loader component', () => {
   it ('should render Loader component', () => {
      const component = setUpRender();
      expect(component).toMatchSnapshot();
   })
});