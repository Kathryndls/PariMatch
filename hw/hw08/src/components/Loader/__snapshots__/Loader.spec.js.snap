// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should render Loader component 1`] = `
<div
  class="loadingio-spinner-dual-ring-cqkmvary9bi"
>
  <div
    class="ldio-lgz85uwngd"
  >
    <div />
    <div>
      <div />
    </div>
  </div>
</div>
`;
